americangut/power_plots.py

```diff
@@ -0,0 +1,581 @@
+    if kwds['show_bound']:
+        uax.set_axis_bgcolor('none')
+    # Makes the lower bound lines dashed and thin, and changes the color if
+    # desired
+    for idx, l in enumerate(uax.get_lines()):
+        l.set_linestyle(':')
+        l.set_linewidth(1.5)
+        if kwds['colormap'] is not None and len(kwds['colormap'].shape) == 1:
+            l.set_color(kwds['colormap'][idx])
+        elif kwds['colormap'] is not None:
+            l.set_color(kwds['colormap'][idx, :])
+    # Hides the x ticks and labels
+    uax.set_title('')
+    uax.set_xticklabels('')
+    uax.set_yticklabels('')
+    uax.set_xlabel('')
+    # Hides the legend
+    uax.get_legend().set_visible(False)
+
+    # Plots the mean data
+    axm = fig.add_axes(lax.get_position())
+    fig = ft.plot_power('nobs', sample_counts, effect_means, ax=axm,
+                        alpha=kwds['alpha'])
+
+    # Shows the confidence bounds, if desired
+    if kwds['show_bound']:
+        axm.set_axis_bgcolor('none')
+
+    # Recolors the lines, if desired
+    if kwds['colormap'] is not None and len(kwds['colormap'].shape) == 1:
+        for idx, l in enumerate(axm.get_lines()):
+            l.set_color(kwds['colormap'][idx])
+    elif kwds['colormap'] is not None:
+        for idx, l in enumerate(axm.get_lines()):
+            l.set_color(kwds['colormap'][idx, :])
+
+    # Sets up the labels
+    axm.set_xticklabels(map(int, axm.get_xticks()), size=kwds['tick_size'])
+    axm.set_yticklabels(axm.get_yticks(), size=kwds['tick_size'])
+    axm.set_xlabel('Number of Observations', size=kwds['label_size'])
+    axm.set_ylabel('Power of the Test', size=kwds['label_size'])
+    axm.set_title(kwds['title'], size=kwds['title_size'])
+
+    # Adds the grid, if desired
+    if kwds['grid']:
+        axm.grid()
+
+    leg = axm.get_legend()
+    # Sets the legend position
+    if kwds['leg_offset'] is not None:
+        leg.set_bbox_to_anchor(kwds['leg_offset'])
+    # Sets up the legend text
+    for idx, txt in enumerate(leg.get_texts()):
+        txt.set_text(labels[idx])
+        txt.set_size(kwds['legend_size'])
+
+    # Returns the figure
+    return fig
+
+
+def trace_bounds(power, counts):
+    """Gets the mean, upper and lower confidence interval for counts
+
+    Parameters
+    ----------
+    power : ndarray
+        An array with multiple power returns where each row corresponds to a
+        permutation run and each column corresponds to a number of samples
+        drawn.
+    counts : ndarray
+        The number of samples drawn per group for each power calculation.
+
+    Returns
+    -------
+        pwr_mean, pwr_lower, pwr_upper : ndarray
+            The mean, upper and lower bounds for the power array.
+
+    """
+
+    pwr_mean = np.nanmean(power, 0)
+    pwr_bound = confidence_bound(power, axis=0)
+
+    # Prevents the confidence interval of being less than 0 or more than 1
+    pwr_lower = np.vstack((pwr_mean - pwr_bound,
```

Comment:
`pwr_lower` returns a 1d array, the same shape as `pwr_mean`, where each position in the array can take on a value [0, 1]. Vertical stacking means that I take the max of each column, not returning a single value.
The other way I can see to do this would be a list comprehension, and that would be a lot slower, from my understanding.